plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -302,8 +302,11 @@ func (r *NodeAuthorizer) hasPathFrom(nodeName string, startingType vertexType, s
 	}
 
 	// Fast check to see if we know of a destination edge
-	if r.graph.destinationEdgeIndex[startingVertex.ID()].has(nodeVertex.ID()) {
-		return true, nil
+	if index := r.graph.destinationEdgeIndex[startingVertex.ID()]; index != nil {
```

Comment:
This will only hold items that had a destination edge and were added to the index. AddVolumeAttachment, SetNodeConfigMap, and AddPod all have places where they add destination edges and do not update the index.

AddPV explicitly doesn't have the ability to add a destination edge:
```
		// since we don't know the other end of the pvc -> pod -> node chain (or it may not even exist yet), we can't decorate these edges with kubernetes node info
```

Unless all added edges are indexed, we can't make the presence of this index short-circuit a DFS fallback

Reply:
The fact that unit tests are passing with this change as well means that not all relationships are well-exercised in the presence of an index

Reply:
Wow, thanks for catching this!

Could you include this test case in your PR?